spec/services/get_products_service_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe GetProductsService do
+  subject(:service) { described_class.new(category_name) }
+
+  let(:category_name) { 'Thức uống'.downcase }
+
+  let!(:parent_category) { create(:category, name: 'Tất cả'.downcase) }
+  let!(:first_category) { create(:category, name: category_name, parent: parent_category) }
+  let!(:second_category) { create(:category, parent: parent_category) }
+
+  let!(:first_product) { create(:product, category: first_category) }
+  let!(:sizes) { create_list(:size, 1, product: first_product) }
+  let!(:toppings) { create_list(:topping, 1, product: first_product) }
+  let!(:tag) { create(:tag, product: first_product) }
+
+  let!(:second_product) { create(:product, category: second_category) }
+  let!(:another_sizes) { create_list(:size, 1, product: second_product) }
```

Comment:
too many creation in db can slow down the test, only need to test necessary thing